xarray/core/accessors.py

```diff
@@ -110,6 +110,47 @@ def _round_field(values, name, freq):
         return _round_series(values, name, freq)
 
 
+def _strftime_through_array(values, date_format):
+    """ Return string formatted values for all items in array """
+    values_unravelled = values.ravel()
+    field_values = np.array([date.strftime(date_format)
+                             for date in values_unravelled])
+
+    return field_values.reshape(values.shape)
+
+
+def _strftime_through_cftimeindex(values, date_format):
+    """Coerce an array of datetime-like values to a CFTimeIndex
+    and access requested datetime component
+    """
+    from ..coding.cftimeindex import CFTimeIndex
+    values_as_cftimeindex = CFTimeIndex(values.ravel())
+
+    field_values = values_as_cftimeindex._strftime(date_format)
+    return field_values.reshape(values.shape)
+
+
+def _strftime_through_series(values, date_format):
+    """Coerce an array of datetime-like values to a pandas Series and
+    apply string formatting
+    """
+    values_as_series = pd.Series(values.ravel())
+    strs = values_as_series.dt.strftime(date_format)
+    return strs.values.reshape(values.shape)
+
+
+def _strftime(values, date_format):
+    if is_np_datetime_like(values.dtype):
+        access_method = _strftime_through_series
+    else:
+        access_method = _strftime_through_array
```

Comment:
Thanks for offering the two options -- I think I slightly prefer the `_strftime_through_cftimeindex` pathway, because it mirrors how the accessor is implemented for other cftime date attributes (and I like the addition of `strftime` as a method of `CFTimeIndex`).

Reply:
Sounds good, removed the array method and cleaned up the CFTimeIndex route.